plugins/Physics/Physics.cpp

```diff
@@ -738,6 +740,73 @@ void Physics::Impl::UpdatePhysics(EntityComponentManager& _ecm)
             return true;
         });
 
+    // Process WorldVelocityCmd
+    _ecm.Each<components::Model, components::WorldVelocityCmd>([&](const Entity& _entity,
+                                                                   const components::Model*,
+                                                                   components::WorldVelocityCmd*
+                                                                       _modelWorldVelocityCmd) {
+        auto modelIt = this->entityModelMap.find(_entity);
+        if (modelIt == this->entityModelMap.end())
+            return true;
+
+        // TODO(addisu) Store the free group instead of searching for it at
+        // every iteration
+        auto freeGroup = modelIt->second->FindFreeGroup();
+
+        // The FreeGroup is created only for floating-base object that do not have any defined
+        // joint between the world and their base
+        if (!freeGroup) {
+            ignwarn << "Failed to find FreeGroup. Linear and angular velocities commands ignored."
+                    << std::endl;
+            return true;
+        }
+
+        // Get canonical link offset
+        auto canonicalLink = _ecm.ChildrenByComponents(_entity, components::CanonicalLink());
+        if (canonicalLink.empty())
+            return true;
+
+        auto canonicalPoseComp = _ecm.Component<components::Pose>(canonicalLink[0]);
+        if (nullptr == canonicalPoseComp)
+            return true;
+
+        ignition::math::Vector3d& modelWorldLinearVelocity = _modelWorldVelocityCmd->Data().linear;
+        ignition::math::Vector3d& modelWorldAngularVelocity =
+            _modelWorldVelocityCmd->Data().angular;
+
+        ignition::math::Pose3d& M_H_B = canonicalPoseComp->Data();
+        ignition::math::Pose3d B_H_M = M_H_B.Inverse();
+        ignition::math::Vector3d& B_o_M = B_H_M.Pos();
+        ignition::math::Quaterniond& B_R_M = B_H_M.Rot();
+
+        {
+            // Express the angular velocity in the frame of the base link
+            ignition::math::Vector3d wBase = B_R_M * modelWorldAngularVelocity;
+            ignwarn << wBase << std::endl;
+
+            // Set the angular velocity of the canonical link
+            freeGroup->SetWorldAngularVelocity(math::eigen3::convert(wBase));
+        }
+
+        {
+            // Skew-symmetrix matrix from 3D vector
+            auto hat = [](const ignition::math::Vector3d& v) -> ignition::math::Matrix3d {
+                return ignition::math::Matrix3d(0, -v[2], v[1], v[2], 0, -v[0], -v[1], v[0], 0);
+            };
+
+            // Express the linear velocity in the frame of the base link
+            ignition::math::Vector3d vBase =
+                B_R_M * modelWorldLinearVelocity + hat(B_o_M) * (B_R_M * modelWorldAngularVelocity);
```

Comment:
Why not 
```suggestion
                B_R_M * modelWorldLinearVelocity + B_o_M.Cross(B_R_M * modelWorldAngularVelocity);
```
That is more compact, see https://ignitionrobotics.org/api/math/4.0/classignition_1_1math_1_1Vector3.html#a304424f098727d9eacc755637addfd05 for the used method.

Reply:
Accordingly to https://github.com/robotology/gym-ignition/pull/117#discussion_r363124317, this transformation is not necessary.